File: Advanced/app/src/main/java/com/bihe0832/request/demo/MainActivity.java
package com.bihe0832.request.demo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.bihe0832.request.demo.advanced.R;
import com.bihe0832.request.demo.request.GetRequest;
import com.bihe0832.request.demo.request.PostRequest;
import com.bihe0832.request.demo.request.TestResponse;
import com.bihe0832.request.libware.request.HTTPServer;
import com.bihe0832.request.libware.request.HttpResponseHandler;

public class MainActivity extends AppCompatActivity {


    private EditText mRequestParaEditText;
    private TextView mResultView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mRequestParaEditText = (EditText)findViewById(R.id.request_paraEditText);
        mResultView = (TextView)findViewById(R.id.request_resultTextView);

        Button getRequestButton = (Button) findViewById(R.id.request_getBtn);
        getRequestButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                sendGetRequest();
            }
        });

        Button postRequestButton = (Button) findViewById(R.id.request_postBtn);
        postRequestButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                sendPostRequest();
            }
        });

        Button clearResultButton = (Button) findViewById(R.id.request_clear);
        clearResultButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                mResultView.setText("");
            }
        });
    }

    private void sendGetRequest(){
        String para = mRequestParaEditText.getText().toString();
        if(null != para && para.length() > 0){
            TestResponesHandler handle = new TestResponesHandler();
            GetRequest request = new GetRequest(para,handle);
            HTTPServer.getInstance().doRequest(request);
        }else{
            showResult("请在输入框输入请求内容！");
        }
    }

    private void sendPostRequest(){
        String para = mRequestParaEditText.getText().toString();
        if(null != para && para.length() > 0){
            TestResponesHandler handle = new TestResponesHandler();
            PostRequest request = new PostRequest(para,handle);
            HTTPServer.getInstance().doRequest(request);
        }else{
            showResult("请在输入框输入请求内容！");
        }
    }

    private class TestResponesHandler implements HttpResponseHandler<TestResponse> {

        @Override
        public void onResponse(TestResponse response) {
            showResult("ret：\n\t" + response.ret + " \n " +
                    "msg：\n\t" + response.msg+ " \n " +
                    "para：\n\t" + response.para+ " \n ");
        }
    }

    private void showResult(final String tips){
        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                mResultView.setText(tips);
            }
        });
    }


}

File: app/src/main/java/com/bihe0832/http/demo/MainActivity.kt
package com.bihe0832.http.demo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.bihe0832.android.http.common.common.HTTPServer
import com.bihe0832.android.http.common.common.HttpBasicRequest
import com.bihe0832.android.http.common.common.HttpResponseHandler
import com.bihe0832.http.advanced.HttpAdvancedRequest
import com.bihe0832.http.demo.request.Constants
import com.bihe0832.http.demo.request.advanced.AdvancedGetRequest
import com.bihe0832.http.demo.request.advanced.AdvancedPostRequest
import com.bihe0832.http.demo.request.advanced.TestResponse
import com.bihe0832.http.demo.request.basic.BasicGetRequest
import com.bihe0832.http.demo.request.basic.BasicPostRequest
import com.bihe0832.request.demo.basic.R
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        getBasic.setOnClickListener { sendGetBasicRequest() }

        postBasic.setOnClickListener { sendPostBasicRequest() }

        getAdvanced.setOnClickListener { sendGetAdvancedRequest() }

        postAdvanced.setOnClickListener { sendPostAdvancedRequest() }

        clearResult.setOnClickListener { result.text = "" }
    }

    private fun showResult(tips: String) {
        runOnUiThread { result.text = tips }
    }

    private fun sendGetBasicRequest() {
        var result = paraEditText.text?.toString()
        if (result?.length ?: 0 > 0) {
            val handle = TestBasicResponseHandler()
            val request = BasicGetRequest(result, handle)
            HTTPServer.getInstance().doRequest(request)
        } else {
            showResult("请在输入框输入请求内容！")
        }
    }

    private fun sendGetAdvancedRequest() {
        var result = paraEditText.text?.toString()
        if (result?.length ?: 0 > 0) {

            HTTPServer.getInstance().doRequest(object : HttpAdvancedRequest<TestResponse>() {
                var res = object : AdvancedResponseHandler<TestResponse> {
                    override fun onRequestSuccess(response: TestResponse) {
                        showResult(response.toString())
                    }

                    override fun onRequestFailure(statusCode: Int, response: String?) {
                        showResult("HTTP状态码：\n\t$statusCode \n 网络请求内容：\n\t$response")
                    }
                }

                override fun getUrl(): String {
                    val builder = StringBuilder()
                    builder.append(Constants.PARA_PARA + HttpBasicRequest.HTTP_REQ_ENTITY_MERGE + result)
                    return Constants.HTTP_DOMAIN + Constants.PATH_GET + "?" + builder.toString()
                }

                override fun getAdvancedResponseHandler(): AdvancedResponseHandler<*> {
                    return res
                }
            })

            AdvancedGetRequest(result, TestAdvancedResponseHandler()).let {
                HTTPServer.getInstance().doRequest(it)
            }
        } else {
            showResult("请在输入框输入请求内容！")
        }
    }

    private fun sendPostBasicRequest() {
        var result = paraEditText.text?.toString()
        if (result?.length ?: 0 > 0) {
            val handle = TestBasicResponseHandler()
            val request = BasicPostRequest(result, handle)
            HTTPServer.getInstance().doRequest(request)
        } else {
            showResult("请在输入框输入请求内容！")

        }
    }

    private fun sendPostAdvancedRequest() {
        var result = paraEditText.text?.toString()

        if (result?.length?:0 > 0) {

            HTTPServer.getInstance().doRequest(object : HttpAdvancedRequest<TestResponse>() {
                init {
                    try {
                        this.data = (Constants.PARA_PARA + HttpBasicRequest.HTTP_REQ_ENTITY_MERGE + result).toByteArray(charset("UTF-8"))
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                }
                var res = object : AdvancedResponseHandler<TestResponse> {
                    override fun onRequestSuccess(response: TestResponse) {
                        showResult(response.toString())
                    }

                    override fun onRequestFailure(statusCode: Int, response: String?) {
                        showResult("HTTP状态码：\n\t$statusCode \n 网络请求内容：\n\t$response")
                    }
                }

                override fun getUrl(): String {
                    return Constants.HTTP_DOMAIN + Constants.PATH_POST
                }

                override fun getAdvancedResponseHandler(): AdvancedResponseHandler<*> {
                    return res
                }
            })

            AdvancedPostRequest(result, TestAdvancedResponseHandler()).let {
                HTTPServer.getInstance().doRequest(it)
            }
        } else {
            showResult("请在输入框输入请求内容！")
        }
    }

    private inner class TestBasicResponseHandler : HttpResponseHandler {

        override fun onResponse(statusCode: Int, response: String) {
            showResult("HTTP状态码：\n\t" + statusCode + " \n " +
                    "网络请求内容：\n\t" + response)
        }
    }

    private inner class TestAdvancedResponseHandler : HttpAdvancedRequest.AdvancedResponseHandler<TestResponse> {
        override fun onRequestSuccess(response: TestResponse) {
            showResult(response.toString())
        }

        override fun onRequestFailure(statusCode: Int, response: String?) {
            showResult("HTTP状态码：\n\t$statusCode \n 网络请求内容：\n\t$response")
        }
    }
}
